index.js

```diff
@@ -187,11 +187,20 @@ function writeShim_ (src, to, opts) {
     '$basedir=Split-Path $MyInvocation.MyCommand.Definition -Parent\n' +
     '\n' +
     '$exe=""\n' +
+    (opts.nodePath ? '$env_node_path=$env:NODE_PATH\n' +
+      `$env:NODE_PATH="${isWindows() ? opts.nodePath : String(opts.nodePath).replace(':', ';')}"\n` : '') +
     'if ($PSVersionTable.PSVersion -lt "6.0" -or $IsWindows) {\n' +
     '  # Fix case when both the Windows and Linux builds of Node\n' +
     '  # are installed in the same directory\n' +
     '  $exe=".exe"\n' +
-    '}\n'
+    '}'
+  if (opts.nodePath && (String(opts.nodePath).includes(';') || String(opts.nodePath).includes(':'))) {
+    pwsh = pwsh +
+      ' else {\n' +
+      `  $env:NODE_PATH="${isWindows() ? opts.nodePath.replace(';', ':') : opts.nodePath}"\n` +
```

Comment:
why is the normalization of PATH needed? pnpm sends in the path correctly, with `:` on non-win and `;` on win

Reply:
This is because the PowerShell script can be executed on both Windows and non-Windows, which it will be on my machine, where I configured WSL (Bash on Ubuntu on Windows) to use `C:/NodeJS/` (`/mnt/c/NodeJS/`) as the Node.js installation directory.

Reply:
pnpm uses the current path delimiter of the OS: https://github.com/pnpm/link-bins/blob/master/src/index.ts#L107

so it should come in correctly

Reply:
Yes, but a PowerShell script generated on Windows won’t work correctly in a Linux/Unix environment (eg. Windows Subsystem for Linux, which allows installing Linux distros from the Windows Store) and vice versa.

Reply:
in that case, I assume `@pnpm/link-bins` should pass an array of paths to `cmd-shim` and cmd-shim should concatenate it the way it needs for ever script

Reply:
Probably, also the script currently produces an invalid `NODE_PATH` for Cygwin/WSL when `pnpm install` is run from windows, and vice versa.